### ISSUE TITLE
interventions- could use an end_day parameter

### ISSUE THREAD

**Issue by cliffckerr** [InstituteforDiseaseModeling/covasim#134]
See issue description here: https://github.com/amath-idm/covasim/issues/317

Per @CWiswell-IDM:
There's probably a useful scenario here where a testing regime starts, and at some point during the epidemic either is cancelled or runs out of funding.

Suggest 

- adding end_day=None to the parameters for test_prob.__init__
- adding something like the following to test_prob.apply
```
if self.end_day:
  if t > end_day:
    return # no more testing now
```

**Comment by wpettine** [comment]
I created my own helper function to do this. Might be worth incorporating somehwere

```
def calc_date_diff(date_1, date_2):
    """Number of days between two dates, year-month-day"""
    dt_1, dt_2 = date_1.split('-'), date_2.split('-')
    dt_1, dt_2 = list(map(int, dt_1)), list(map(int, dt_2))
    delta = date(dt_2[0], dt_2[1], dt_2[2]) - date(dt_1[0], dt_1[1], dt_1[2])
    return delta.days
```

Thus, when setting the number of days, I do something to the effect of: 
```
start_day = '2020-02-01'
end_day = '2020-05-15'

n_days = calc_date_diff(start_day, end_day)
```

**Comment by ckerr-IDM** [comment]
Thanks @wpettine -- to handle different date formats, I suggest `sciris.readdate()`, i.e.
```
    dt_1, dt_2 = date_1.split('-'), date_2.split('-')
    dt_1, dt_2 = list(map(int, dt_1)), list(map(int, dt_2))
    delta = date(dt_2[0], dt_2[1], dt_2[2]) - date(dt_1[0], dt_1[1], dt_1[2])
```
would just be
```
    dt_1, dt_2 = sc.readdate(date_1), sc.readdate(date_2)
    dt_1, dt_2 = list(map(int, dt_1)), list(map(int, dt_2))
    delta = dt_2 - dt_1
```

**Comment by lgeorge-idm** [comment]
@wpettine, currently some interventions don't allow for setting a range of days with both a start and end.
That still needs to be implemented.

**Comment by cliffckerr** [comment]
Closed by #171
